grails-app/services/org/pih/warehouse/core/DashboardService.groovy

```diff
@@ -554,6 +554,50 @@ class DashboardService {
         return reorderStock
     }
 
+    def getReorderReport(Location location) {
+        long startTime = System.currentTimeMillis()
+        ArrayList<Map> inventoryItems = getInventoryItems(location)
+
+        ArrayList<Map> reorderStock = inventoryItems.findAll { Map inventoryItem ->
+            def inventoryLevel = inventoryItem?.inventoryLevel as InventoryLevel
```

Comment:
This is expensive, so I would consider doing this using a join in the getInventoryItems method. That's tricky because you have to do some domain mapping in the criteria query (or HQL).

Also getInventoryItems should probably be changed to getAvailableItems and should return the AvailableItem instead. Not a big deal for now, but we should clean that up when we get a chance.